activity/worker.go

```diff
@@ -0,0 +1,172 @@
+package activity
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/awslabs/aws-sdk-go/gen/swf"
+	"github.com/juju/errors"
+	"github.com/sclasen/swfsm/fsm"
+	"github.com/sclasen/swfsm/poller"
+	. "github.com/sclasen/swfsm/sugar"
+)
+
+type SWFOps interface {
+	RecordActivityTaskHeartbeat(req *swf.RecordActivityTaskHeartbeatInput) (resp *swf.ActivityTaskStatus, err error)
+	RespondActivityTaskCanceled(req *swf.RespondActivityTaskCanceledInput) (err error)
+	RespondActivityTaskCompleted(req *swf.RespondActivityTaskCompletedInput) (err error)
+	RespondActivityTaskFailed(req *swf.RespondActivityTaskFailedInput) (err error)
+	PollForActivityTask(req *swf.PollForActivityTaskInput) (resp *swf.ActivityTask, err error)
+}
+
+type ActivityWorker struct {
+	Serializer fsm.StateSerializer
+	// Domain of the workflow associated with the FSM.
+	Domain string
+	// TaskList that the underlying poller will poll for decision tasks.
+	TaskList string
+	// Identity used in PollForActivityTaskRequests, can be empty.
+	Identity string
+	// Client used to make SWF api requests.
+	SWF SWFOps
+	// Type Info for handled activities
+	handlers map[string]*ActivityHandler
+	// ShutdownManager
+	ShutdownManager *poller.ShutdownManager
+	// ActivityTaskDispatcher
+	ActivityTaskDispatcher ActivityTaskDispatcher
+	// ActivityInterceptor
+	ActivityInterceptor ActivityInterceptor
+	// allow panics in activities rather than recovering and failing the activity, useful for testing
+	AllowPanics bool
+}
+
+func (a *ActivityWorker) AddHandler(handler *ActivityHandler) {
+	if a.handlers == nil {
+		a.handlers = map[string]*ActivityHandler{}
+	}
+	a.handlers[handler.Activity] = handler
+}
+
+func (a *ActivityWorker) Init() {
```

Comment:
I'd much rather see an `activity.New(required...) *activityActivityWorker` which sets defaults for all fields (e.g.: `exec.Command(...) *exec.Cmd`).